internal/objects/validator.go

```diff
@@ -165,7 +165,7 @@ func (v *Validator) handle(ctx context.Context, req *request) admission.Response
 
```

Comment:
I'm not sure about this at all. 

The check was to prevent desendents being overwritten, *but* once the above fix is in place to add source objects for descendants this validation check fails even if you're just updating the parent object. I flipped it to ancestors just so flux stopped getting dry-run failures in our environment.

I'm not sure the check makes much sense if validating against the forest because if the forest contains source objects, etc., already, you've got to assume HNC created them and any conflicts should have caused errors earlier in the process.